WordPress/Classes/Stores/PluginStore.swift

```diff
@@ -192,6 +243,27 @@ class PluginStore: QueryStore<PluginStoreState, PluginQuery> {
         }
     }
 
+    private func reinitialiseIfNeeded() {
```

Comment:
I think this method gets a bit confusing. I see I did a similar thing on `queriesChanged` updating the state with a transaction, but technically the state of the store hasn't changed now, we're just dropping it from memory, and it should be the same when we access it again (by adding new queries).

I have a couple ideas around this:
1. Make a state frontend to deal with persistence. This can be as simple as an Obj-C style lazy property (make the in-memory state a private optional property, and have `state` either return or initialize state).
2. See if we can make persistence a part of `QueryStore`/`StatefulStore`